server/src/main/java/org/apache/cassandra/sidecar/acl/authorization/RoleBasedAuthorizationProvider.java

```diff
@@ -64,29 +59,28 @@ public void getAuthorizations(User user, Handler<AsyncResult<Void>> handler)
     @Override
     public Future<Void> getAuthorizations(User user)
     {
-        List<String> identities = extractIdentities(user);
+        List<String> roles = extractCassandraRoles(user);
 
-        if (identities.isEmpty())
+        if (roles.isEmpty())
         {
-            return Future.failedFuture("Missing client identities");
+            return Future.failedFuture("No cassandra roles found associated with the user");
         }
 
-        Set<Authorization> authorizations = new HashSet<>();
-        for (String identity : identities)
+        for (String role : roles)
```

Comment:
can role be null? I'm referring to the check in line 71.  I guess it's fine to have the check, but it feels that it would be an error to return null

Reply:
We shouldn't get a null entry in List from roles extracted from user attributes, we discussed to keep this check for safer side